src/videodownlinkbandwidthpolicy/AllHighestVideoBandwidthPolicy.ts

```diff
@@ -61,26 +62,41 @@ export default class AllHighestVideoBandwidthPolicy implements VideoDownlinkBand
       this.selfAttendeeId
     );
 
-    // If video sources are not chosen, then return the default receive set.
-    if (this.videoSources === undefined) {
-      return receiveSet;
-    }
+    // If video sources are not chosen, then use all of them.
+    const videoSources = !!this.videoSources
+      ? this.videoSources
+      : this.videoIndex.allVideoSendingSourcesExcludingSelf(this.selfAttendeeId);
 
     // Get the list of all the remote stream information
     const remoteInfos = this.videoIndex.remoteStreamDescriptions();
 
-    const mapOfAttendeeIdToOptimalStreamId = new Map<string, number>();
+    const mapOfAttendeeIdToRemoteDescriptions = new Map<string, VideoStreamDescription>();
 
     for (const info of remoteInfos) {
       if (receiveSet.contain(info.streamId)) {
-        mapOfAttendeeIdToOptimalStreamId.set(info.attendeeId, info.streamId);
+        mapOfAttendeeIdToRemoteDescriptions.set(info.attendeeId, info);
       }
     }
 
-    for (const videoSource of this.videoSources) {
+    let totalBitrateKbps = 0;
+    for (const videoSource of videoSources) {
       const attendeeId = videoSource.attendee.attendeeId;
-      if (mapOfAttendeeIdToOptimalStreamId.has(attendeeId)) {
-        streamSelectionSet.add(mapOfAttendeeIdToOptimalStreamId.get(attendeeId));
+      if (mapOfAttendeeIdToRemoteDescriptions.has(attendeeId)) {
+        const info = mapOfAttendeeIdToRemoteDescriptions.get(attendeeId);
+        // Cap total receive bitrate at 15000 kbps
+        if (totalBitrateKbps + info.maxBitrateKbps <= 15000) {
```

Comment:
Default bitrate is like 200kbps at 10 videos. It runs into TURN limits above 20Mbps, which is why we added this, since SVC will always send a max layer with a high bitrate. But if that is the case they should be using some sort of video stream selection (manual/through priority policy).